### DISCUSSION TITLE
about collusion track

### DISCUSSION THREAD

**Discussion by rezunli96** [yasserfarouk/scml#23] [collusion-qa] [answered]
Can an agent know which factories they are controlling in the collusion track? From my reading of the code it appears that in each simulation it first assigns multiple factories by the same python agent class, and then treat them as independent agents.

**Comment by yasserfarouk** [comment, marked as answer]
You can do that by keeping a class variable and updating it in your `init` method.

As an example

```python
class MyAgent:
    my_brothers = []

    def init():
         MyAgent.my_brothers.append(self)
```
